e2e-tests/src/test/staking.rs

```diff
@@ -172,47 +149,64 @@ fn get_key_pairs() -> (Vec<KeyPair>, Vec<KeyPair>) {
+// 3. bond controller account to stash account, stash = controller and set controller to StakerStatus::Nominate
+// 4. wait for new era
+// 5. send payout stakers tx
 pub fn staking_test(config: &Config) -> anyhow::Result<()> {
     const TOKEN: u128 = 1_000_000_000_000;
-    const VALIDATOR_STAKE: u128 = 25_000u128 * TOKEN;
-    const NOMINATOR_STAKE: u128 = 1_000u128 * TOKEN;
+    const VALIDATOR_STAKE: u128 = 25_000 * TOKEN;
+    const NOMINATOR_STAKE: u128 = 1_000 * TOKEN;
 
-    let node = &config.node;
     let (stashes_accounts, validator_accounts) = get_key_pairs();
-    endow_stash_balances(
-        node,
-        validator_accounts[0].clone(),
-        &stashes_accounts,
-        VALIDATOR_STAKE,
-    );
 
-    for validator_account in validator_accounts.clone() {
-        let validator_account = validator_account.to_owned();
-        bond(node, VALIDATOR_STAKE, validator_account.clone());
-        validate(node, validator_account);
-    }
+    let node = &config.node;
+    let sender = validator_accounts[0].clone();
+    let connection = create_connection(node).set_signer(sender);
 
-    for (nominator, nominee) in stashes_accounts
-        .into_iter()
-        .zip(validator_accounts.clone().into_iter())
-    {
-        let nominator = nominator.to_owned();
-        let nominee = nominee.to_owned();
-        bond(node, NOMINATOR_STAKE, nominator.clone());
-        nominate(node, nominator, nominee);
-    }
+    endow_stash_balances(&connection, &stashes_accounts, VALIDATOR_STAKE);
 
-    let sender = validator_accounts[0].clone();
-    let current_era = wait_for_full_reward_era(node, sender.clone())?;
+    validator_accounts.par_iter().for_each(|account| {
+        bond(node, VALIDATOR_STAKE, account);
+    });
+
+    validator_accounts
+        .par_iter()
+        .for_each(|account| validate(node, account));
+
+    stashes_accounts
+        .par_iter()
+        .for_each(|nominator| bond(node, NOMINATOR_STAKE, nominator));
+
+    stashes_accounts
+        .par_iter()
+        .zip(validator_accounts.par_iter())
+        .for_each(|(nominator, nominee)| nominate(node, nominator, nominee));
+
+    let current_era = wait_for_full_reward_era(&connection)?;
     info!(
-        "[+] Full era {} passed, claiming rewards for era {}",
+        "Full era {} passed, claiming rewards for era {}",
         current_era,
         current_era - 1
     );
 
-    for validator_account in validator_accounts {
-        payout_staker(node, sender.clone(), &validator_account, current_era - 1);
-    }
+    validator_accounts
+        .into_par_iter()
+        .for_each(|account| payout_stakers(node, account, current_era - 1));
+
+    // Sanity check
+    let block_number = connection
+        .get_header::<Header>(None)
+        .unwrap()
+        .unwrap()
+        .number;
+    info!(
+        "Current block number is {}, waiting till it finalizes",
+        block_number,
+    );
+
+    wait_for_finalized_block(&connection, block_number);
```

Comment:
This line raises a warning , please consider fixing it
```
warning: unused `Result` that must be used
   --> src/test/staking.rs:209:5
    |
209 |     wait_for_finalized_block(&connection, block_number);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_must_use)]` on by default
    = note: this `Result` may be an `Err` variant, which should be handled
```

Reply:
fixed